libs/clients/cbr/client_test.go

```diff
@@ -102,3 +109,183 @@ func TestSignAndRedeemCredentials(t *testing.T) {
 	err = client.RedeemCredentials(ctx, []CredentialRedemption{{Issuer: issuerName, TokenPreimage: preimage, Signature: sig}}, payload)
 	assert.NoError(t, err, "Should be able to bulk redeem tokens")
 }
+
+func TestCreateIssuerV3(t *testing.T) {
+	client, err := New()
+	assert.NoError(t, err)
+
+	request := IssuerRequest{
+		Name:      test.RandomString(),
+		Cohort:    int16(test.RandomNonZeroInt(10)),
+		MaxTokens: test.RandomNonZeroInt(10),
+		ValidFrom: ptr.FromTime(time.Now()),
+		Duration:  "P1M",
+		Buffer:    test.RandomNonZeroInt(10),
+		Overlap:   test.RandomNonZeroInt(10),
+	}
+
+	err = client.CreateIssuerV3(context.Background(), request)
+	assert.NoError(t, err)
+}
+
+func TestGetIssuerV2(t *testing.T) {
+	ctx := context.Background()
+
+	client, err := New()
+	assert.NoError(t, err)
+
+	issuerRequest := IssuerRequest{
+		Name:      test.RandomString(),
+		Cohort:    int16(test.RandomNonZeroInt(10)),
+		MaxTokens: test.RandomNonZeroInt(10),
+		ValidFrom: ptr.FromTime(time.Now()),
+		ExpiresAt: ptr.FromTime(time.Now().Add(time.Hour)),
+		Duration:  "P1M",
+		Buffer:    test.RandomNonZeroInt(30),
+		Overlap:   test.RandomNonZeroInt(10),
+	}
+
+	err = client.CreateIssuerV3(ctx, issuerRequest)
+	assert.NoError(t, err)
+
+	issuer, err := client.GetIssuerV2(ctx, issuerRequest.Name, issuerRequest.Cohort)
+	assert.NoError(t, err)
+
+	assert.Equal(t, issuerRequest.Name, issuer.Name)
+	assert.Equal(t, issuerRequest.Cohort, issuer.Cohort)
+	assert.Equal(t, issuerRequest.ExpiresAt.Format(time.RFC3339), issuer.ExpiresAt)
+	assert.NotEmpty(t, issuer.PublicKey)
+}
+
+func TestSignAndRedeemCredentialsV3(t *testing.T) {
+	databaseURL := os.Getenv("CHALLENGE_BYPASS_DATABASE_URL")
+
+	sKey := "fzJbqh6l/xWAjT6Ulmu+/Taxz8XZ7SDnJ/dUXPgtnQE="
+	blindedToken := "yoGo7zfMr5vAzwyyFKwoFEsUcyUlXKY75VvWLfYi7go="
+	signedToken := "ohwnBITMSphAFK/06LtbC+PYl6PmmEhOdybvsfqZjG4="
+	preimage := "Aa61pQzyxsy3Z6tSwccnOqiW23fNYp0z3xw6XGlA5FG8O/EqlxR87DWnas49U2JUau44dpiveAt7kBXDH5RjPQ=="
+	sig := "zx1zdMhN4Et8WnrkVQOad6xhUBAJ7Pq4b8A0n96CRE0QdAQ+tJe0/eFiJqIPMuKkyfQ6VncIkGj9VzkByh9uFA=="
+	payload := "test message"
+
+	db, err := sql.Open("postgres", databaseURL)
+	if err != nil {
+		assert.NoError(t, err, "Must be able to connect to challenge-bypass db")
+	}
+
+	_, err = db.Exec("DELETE from v3_issuer_keys; DELETE FROM v3_issuers; DELETE from redemptions")
```

Comment:
oof, I don't love us having to muck around in the challenge bypass database as part of our tests :(

Reply:
we could expose a protected endpoint in CB to safely remove an issuer and associated keys, would this be preferred?

Reply:
This is how the original tests prior to this work were done and agreed its not great. I would argue we shouldn't be calling cbr at all and use wiremock and self initialise stubs or something similar. Essentially we are testing cbr implementation here.

Reply:
tracking here: https://github.com/brave-intl/bat-go/issues/1702 ... Since this is legacy paradigm I think we can fix this behavior in another task.

Reply:
tracking here: https://github.com/brave-intl/bat-go/issues/1702 ... Since this is legacy paradigm I think we can fix this behavior in another task.